### PR TITLE
Separate operator and edge-api codebases

### PR DESCRIPTION
**Pull request by jakub-dzon**
This PR separates existing operator and edge-api code to allow for easier future changes and better Dockerfile-level separation.

No functional code changes, just scattering `internal/*` among new `internal/common`, `internal/operator` and `internal/edgeapi`.

### CODE REVIEW

File: internal/edgeapi/yggdrasil/yggdrasil_integration_test.go [hunk 1]
```diff
@@ -29,16 +29,16 @@ import (
...
+	"github.com/project-flotta/flotta-operator/internal/common/metrics"
+	"github.com/project-flotta/flotta-operator/internal/common/repository/edgedevice"
+	"github.com/project-flotta/flotta-operator/internal/common/repository/edgedeviceset"
+	"github.com/project-flotta/flotta-operator/internal/common/repository/edgedevicesignedrequest"
+	"github.com/project-flotta/flotta-operator/internal/common/repository/edgeworkload"
+	"github.com/project-flotta/flotta-operator/internal/edgeapi/configmaps"
+	"github.com/project-flotta/flotta-operator/internal/edgeapi/devicemetrics"
+	"github.com/project-flotta/flotta-operator/internal/edgeapi/images"
+	"github.com/project-flotta/flotta-operator/internal/edgeapi/k8sclient"
+	yggdrasil2 "github.com/project-flotta/flotta-operator/internal/edgeapi/yggdrasil"
```
Comment (masayag) [inline internal/edgeapi/yggdrasil/yggdrasil_integration_test.go:41]: why yggdrasil2 ?

File: main.go [hunk 2]
```diff
@@ -18,21 +18,21 @@ package main
 
 import (
 	"context"
+	"github.com/project-flotta/flotta-operator/internal/common/indexer"
```
Comment (masayag) [inline main.go:21]: the formatting of the packages isn't the desired one.
pls check the output of `make all` and compare

**Review by masayag — CHANGES_REQUESTED** [review]

File: controllers/edgedevice_controller.go [hunk 3]
```diff
@@ -18,6 +18,10 @@ package controllers
 
 import (
 	"context"
+	"github.com/project-flotta/flotta-operator/internal/common/metrics"
```
[the remark this answers is not in the document]
Comment (jakub-dzon) [reply controllers/edgedevice_controller.go]: `make imports` did that to me... Something is not right with it on my machine.

File: main.go (hunk 2 above)
Comment (jakub-dzon) [reply main.go:28]: For some reason `make all` re-formats almost all of the imports and it is different from what's in `main`.

Comment (jakub-dzon) [reply main.go:28]: Fixed the file

**Comment by jakub-dzon** [comment]
@gabriel-farache the distinction between the packages is very similar to what you described, with following comments:
 - `edgeapi` is called this way, because it holds files used only by flotta-edge-api container/pod/deployment; the API for communicating with edge devices is called Flotta Edge API throughout the docs;
 - `operator` in turn holds files used exclusively by the operator;
 - `common` - code shared between both binaries.

All three packages can potentially become separate github repos/projects.

**Comment by jakub-dzon** [comment]
`common` is common for operator _binaries_ (so also `/controllers` package).
If there is a separate subpackage for edgeapi, separate one for operator seems consistent.

**Review by masayag — APPROVED** [review]

**Comment by gabriel-farache** [comment]
> `common` is common for operator _binaries_ (so also `/controllers` package). If there is a separate subpackage for edgeapi, separate one for operator seems consistent.
If `common` is used by packages outside `internal` should it be in the root folder? 
Well, as far as I understand it, `edgeapi` interaction is part of the responsbilities of the operator, hence a subpackage for this particular responsibiliy. But naming a subpackage `operator` inside the operator project is a pleonasm, this is what is bugging me.
I know I am quibbling about it and if I am the only one 'annoyed' about it, it is fine and we should move forward with your naming :)

**Comment by masayag** [comment]
> If `common` is used by packages outside `internal` should it be in the root folder? Well, as far as I understand it, `edgeapi` interaction is part of the responsbilities of the operator, hence a subpackage for this particular responsibiliy. But naming a subpackage `operator` inside the operator project is a pleonasm, this is what is bugging me. I know I am quibbling about it and if I am the only one 'annoyed' about it, it is fine and we should move forward with your naming :)

`edgeapi` can be provided and run outside of the context of the operator, even on a different cluster as a service that points to the cluster on which the operator is deployed, or even as a different deployment method (e.g. not on top of k8s - not yet supported, but it doable). Therefore next step might be splitting this repository for the operator and for the `edgeapi` (web-server).

This repository will include flotta's CRDs that should be imported and used by `edgeapi`. I'm okay with the current split and renaming, looking forward for additional work on top of it.

Once repo is split, there will no longer be a need for the operator/edgeapi folders.

**Comment by gabriel-farache** [comment]
> > If `common` is used by packages outside `internal` should it be in the root folder? Well, as far as I understand it, `edgeapi` interaction is part of the responsbilities of the operator, hence a subpackage for this particular responsibiliy. But naming a subpackage `operator` inside the operator project is a pleonasm, this is what is bugging me. I know I am quibbling about it and if I am the only one 'annoyed' about it, it is fine and we should move forward with your naming :)
> 
> `edgeapi` can be provided and run outside of the context of the operator, even on a different cluster as a service that points to the cluster on which the operator is deployed, or even as a different deployment method (e.g. not on top of k8s - not yet supported, but it doable). Therefore next step might be splitting this repository for the operator and for the `edgeapi` (web-server).
> 
> This repository will include flotta's CRDs that should be imported and used by `edgeapi`. I'm okay with the current split and renaming, looking forward for additional work on top of it.
> 
> Once repo is split, there will no longer be a need for the operator/edgeapi folders.

Ok, so this is just a step toward another split

**Comment by jakub-dzon** [comment]
> > > If `common` is used by packages outside `internal` should it be in the root folder? Well, as far as I understand it, `edgeapi` interaction is part of the responsbilities of the operator, hence a subpackage for this particular responsibiliy. But naming a subpackage `operator` inside the operator project is a pleonasm, this is what is bugging me. I know I am quibbling about it and if I am the only one 'annoyed' about it, it is fine and we should move forward with your naming :)
> > 
> > 
> > `edgeapi` can be provided and run outside of the context of the operator, even on a different cluster as a service that points to the cluster on which the operator is deployed, or even as a different deployment method (e.g. not on top of k8s - not yet supported, but it doable). Therefore next step might be splitting this repository for the operator and for the `edgeapi` (web-server).
> > This repository will include flotta's CRDs that should be imported and used by `edgeapi`. I'm okay with the current split and renaming, looking forward for additional work on top of it.
> > Once repo is split, there will no longer be a need for the operator/edgeapi folders.
> 
> Ok, so this is just a step toward another split

It is step forward more refactoring and maybe a repo split in the future.

**Review by machacekondra — APPROVED** [review]

**Pull request merged**